docs/rules/string-match.md

```diff
@@ -0,0 +1,44 @@
+# disallow strings that match a given regular expression (string-match)
+
+Strings are gold mines of hidden information: I've seen type systems and enumerations implemented in strings. I've seen numbers being used as strings to get around type conversions. And sometimes, you just want to enforce some consistency in your errors. This rule allows you to highlight these issues and decide how you want to act on them.
```

Comment:
This might just be me, but I feel like even after reading this paragraph, I still don't understand what this rule would be useful for. Maybe this paragraph could give an example of a scenario where this rule might be useful?